packages/tools/src/tools/segmentation/strategies/fillRectangle.ts

```diff
@@ -30,55 +17,11 @@ function fillRectangle(
   operationData: OperationData,
   inside = true
 ): void {
-  const {
-    volume: segmentation,
-    points,
-    segmentsLocked,
-    segmentIndex,
-    segmentationId,
-    constraintFn,
-  } = operationData;
-  const { imageData, dimensions } = segmentation;
-  const scalarData = segmentation.getScalarData();
-
-  let rectangleCornersIJK = points.map((world) => {
-    return transformWorldToIndex(imageData, world);
-  });
-
-  // math round
-  rectangleCornersIJK = rectangleCornersIJK.map((point) => {
-    return point.map((coord) => {
-      return Math.round(coord);
-    });
-  });
-
-  const boundsIJK = getBoundingBoxAroundShape(rectangleCornersIJK, dimensions);
-
-  if (boundsIJK.every(([min, max]) => min !== max)) {
-    throw new Error('Oblique segmentation tools are not supported yet');
+  if (operationData.editData.type === 'volume') {
+    volumeStrategy.fillRectangle(enabledElement, operationData);
+  } else {
+    stackStrategy.fillRectangle(enabledElement, operationData);
```

Comment:
These operations share a lot of logic. Only branch out the logic for the scalarData that is needed, they share almost 80% of the code